spacerace-client/src/main.rs

```diff
@@ -0,0 +1,302 @@
+async fn run_game_loop(
+    client: reqwest::Client,
+    player_password: String,
+    host: String,
+) -> anyhow::Result<()> {
+    let mut stdout = stdout();
+
+    loop {
+        // Get current game state
+        let state_response = client
+            .get(format!("{}/state", host))
+            .send()
+            .await?
+            .error_for_status()?
+            .json::<GameStateResponse>()
+            .await?;
+
+        // Clear the screen each loop and move cursor to top-left
+        execute!(
+            stdout,
+            terminal::Clear(terminal::ClearType::All),
+            cursor::MoveTo(0, 0)
+        )?;
+
+        // Reprint instructions so they stay visible
+        println!("Use arrow keys (or WASD) to control the ship. Press 'q' to quit.");
+
+        // Determine thrust and rotation based on user input
+        let thrust_rotation = if let GameStateResponse::Active(active) = &state_response {
+            if active.state != "Active" && active.state != "Queued" {
+                debug!(
+                    "Game is not in an active state (state: {}). Press 'q' to quit.",
+                    active.state
+                );
+                thrust_rotation_from_input_with_timeout(Duration::from_millis(200))?
+            } else {
+                // Game is active or queued, allow user to control
+                thrust_rotation_from_input_with_timeout(Duration::from_millis(200))?
+            }
+        } else {
+            debug!("No active game. Press 'q' to quit.");
+            thrust_rotation_from_input_with_timeout(Duration::from_millis(200))?
+        };
+
+        // If user provided control input, send it to server
+        if let Some((thrust, rotation)) = thrust_rotation {
+            let control_req = ControlRequest {
+                password: player_password.clone(),
+                thrust,
+                rotation,
+            };
+
+            let control_resp: ControlResponse = client
+                .post(format!("{}/control", host))
+                .json(&control_req)
+                .send()
+                .await?
+                .error_for_status()?
+                .json()
+                .await?;
+
+            debug!("Control Response: {:?}", control_resp);
+        }
+
+        // Output the game state
+        if let GameStateResponse::Active(active) = &state_response {
+            for ship in &active.ships {
+                println!(
+                    "Ship ID: {}, Position: ({:.1}, {:.1}), Velocity: ({:.1}, {:.1}), Orientation: {:.1}",
+                    ship.id,
+                    ship.position[0],
+                    ship.position[1],
+                    ship.velocity[0],
+                    ship.velocity[1],
+                    ship.orientation
+                );
+            }
+        }
+
+        stdout.flush()?;
+        // Throttle updates slightly to avoid high CPU usage
+        sleep(Duration::from_millis(100)).await;
```

Comment:
Is 100ms short enough to allow for a reasonable response to keyboard events? Seems a bit long.

Reply:
Good point, I'll decrease to 30ms